task-sdk/src/airflow/sdk/io/path.py

```diff
@@ -60,6 +60,9 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
+    def __iter__(self):
+        return iter(self._obj)
+
```

Comment:
Delegating iteration to self._obj makes it work seamlessly with Python’s iteration logic. You can also add a short comment/note above explaining why this delegation is needed, so that it helps people who work on this in the future.

Reply:
Thanks for the review, sending the suggestion
```suggestion
    # We need to explicitly implement `__iter__`
    # because otherwise python iteration logic could use `__getitem__`
    def __iter__(self):
        return iter(self._obj)

```

Reply:
Looks good to me. Thanks for addressing that quickly!